src/clients/BaseAbstractClient.ts

```diff
@@ -72,7 +74,7 @@ export abstract class BaseAbstractClient {
       if (provider instanceof providers.Provider) {
         to = await provider.getBlockNumber();
       } else {
-        const { slot } = await getNearestSlotTime(provider);
+        const { slot } = await getNearestSlotTime(provider, this.logger);
```

Comment:
In general I think we should be doing EVM- and SVM-specific implementations of `updateSearchConfig()` downstream of this class, but that's for follow-up refactoring.

Reply:
Actually, I had to go with this implementation of two different `updateSearchConfig` functions in the `BaseAbstractClient` because you cannot have multiple inheritance in TS and I couldn't get the `Evm/SvmSpokePoolClient` to work because the `SpokePoolClient` needs to extend `BaseAbstractClient`.:

https://github.com/across-protocol/sdk/pull/1171/commits/ae2ff12b62c0d00f428e2205c8885a65b82b6cb7

Reply:
I looked into mixins but they have tradeoffs and the resultant hybrid class isn't a normal class, I'd rather not use mixins